catalog/service/rest/src/main/java/org/projectnessie/catalog/service/rest/ExternalBaseUri.java

```diff
@@ -88,6 +88,24 @@ default void icebergConfigOverrides(BiConsumer<String, String> config) {
     config.accept("uri", icebergBaseURI().toString());
   }
 
+  /**
+   * The path, without a leading slash, of the URI to sign a request to S3 for a specific content
+   * key.
+   *
+   * <p>Must use both {@code s3.signer.uri} and {@code s3.signer.endpoint}, because Iceberg before
+   * 1.5.0 does not handle full URIs passed via {@code s3.signer.endpoint}. This was changed via <a
+   * href="https://github.com/apache/iceberg/pull/8976/files#diff-1f7498b6989fffc169f7791292ed2ccb35b305f6a547fd832f6724057c8aca8bR213-R216">this
+   * Iceberg PR</a> first released in Iceberg 1.5.0. It's unclear how other language implementations
+   * deal with this.
+   *
+   * @param prefix the prefix of the request (warehouse and reference)
+   * @param contentKeyPathString the content key to sign the request for
+   * @param signerParam the signer params path parameter (URL safe representation)
+   */
+  default String icebergS3SignerPathWithPath(String prefix, String signerParam) {
+    return format("v1/%s/s3-sign2/%s", encode(prefix, UTF_8), signerParam);
```

Comment:
Why do we encode `prefix` but not `signerParam`? It looks odd that encoding assumptions are different for these two input parameters.

IMHO, a "URL-safe representation" is not crystal clear. E.g. when constructing ` java.net.URI` from components, it will encode special chars in the provided path string, and on parsing that path element back it will decode those change and we get the original string. Since we assume `signedParams` in `IcebergApiV1S3SignResource` is URL-decoded (by the Web container), I tend to think we have to apply the encoding here, even  though it will probably not change the string in runtime.

Reply:
Base64 using URL-alphabet is safe